x-pack/plugins/apm/server/lib/service_map/get_service_anomalies.ts

```diff
@@ -8,27 +8,40 @@ import { leftJoin } from '../../../common/utils/left_join';
 import { Job as AnomalyDetectionJob } from '../../../../ml/server';
 import { PromiseReturnType } from '../../../typings/common';
 import { IEnvOptions } from './get_service_map';
-import { APM_ML_JOB_GROUP_NAME } from '../../../common/ml_job_constants';
+import {
+  APM_ML_JOB_GROUP_NAME,
+  encodeForMlApi,
+} from '../../../common/ml_job_constants';
 
 type ApmMlJobCategory = NonNullable<ReturnType<typeof getApmMlJobCategory>>;
-const getApmMlJobCategory = (
+export const getApmMlJobCategory = (
   mlJob: AnomalyDetectionJob,
   serviceNames: string[]
 ) => {
+  const serviceByGroupNameMap = new Map(
+    serviceNames.map((serviceName) => [
+      encodeForMlApi(serviceName),
+      serviceName,
+    ])
+  );
   const apmJobGroups = mlJob.groups.filter(
     (groupName) => groupName !== APM_ML_JOB_GROUP_NAME
   );
   if (apmJobGroups.length === mlJob.groups.length) {
```

Comment:
I think this would read better as:

```ts
  const hasApmGroup = mlJob.groups.some(
    (groupName) => groupName !== APM_ML_JOB_GROUP_NAME
  );
  if (!hasApmGroup) {
    // ML job missing "apm" group
    return;
  }
```